### PR TITLE
Ansible server details & skeleton of infrastructure docs needed

### LINKED ISSUE #4: create molecule scenarios / integration tests for playbooks w/o local roles?

### PR DESCRIPTION
**Pull request by cmharlow**
Closes #4 #5

### CODE REVIEW

File: general/ansible-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+- Where appropriate, TUL role releases should be pushed back to Ansible Galaxy.
+- All roles should have the following:
+  - a `.python-version` file with a pinned Python version (preferably, 3.7)
+  - a `Pipfile` and a `Pipfile.lock` with pinned requirements (at least, `ansible` and `molecule`)
+  - All roles should be kept in separate Github repositories (or pulled from Galaxy), and leverage GitHub releases for various Playbooks to be pin versions and reference in their `required-roles.yml`.
+
+We opt to use external roles (instead of writing our own) from GitHub or Ansible Galaxy then the role repository has suitable test coverage, recent activity, documentation, and licensing.
+
+*Proposed:
+  - generate a fork of the ansible-galaxy roles generator for consistent generation of new TUL roles;
```
Comment (bibliotechy) [inline general/ansible-best-practices.md]: Might we want to use the `molecule init` to generate roles, as it creates the scenarios, and allows for creation to be based on a template? https://molecule.readthedocs.io/en/latest/usage.html#init

Comment (cmharlow) [reply general/ansible-best-practices.md]: good idea - happy to add this instead of using `ansible-galaxy`

File: infra/ansible-server.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Ansible for Baseline Server Provisioning & Orchestration
+
+## About
+
+**Overall create workflow / "you are here":**
+1. Dev will manually generate either a [TUCloud](resources.md#tucloud) or [Linode](resources.md#linode-gui) infrastructure resource (usually, a VM box).
+    i. *Proposed: Dev will create PR on Linode Terraform repository to perform CRUD on infrastructure resources; merging triggers change in Linode*
+2. Dev will manually copy / record the IP address from the generated resources.
+    i. *Proposed: Terraform success changes dynamically-updated linode server inventory, and Ansible uses dynamic calls to Ansible API & automated resource tagging to get latest inventory.*
+3. Dev will on local machine run a [server provisioning Ansible playbook](#playbooks) for Linode resources, passing in the new box's IP address as a parameter on the command. For TUCloud resources, these are provisioned manually (Chad to confirm?).
```
Comment (bibliotechy) [inline infra/ansible-server.md]: Confirming that Tucloud resources are crated manually

File: infra/ansible-server.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Ansible for Baseline Server Provisioning & Orchestration
+
+## About
+
+**Overall create workflow / "you are here":**
+1. Dev will manually generate either a [TUCloud](resources.md#tucloud) or [Linode](resources.md#linode-gui) infrastructure resource (usually, a VM box).
+    i. *Proposed: Dev will create PR on Linode Terraform repository to perform CRUD on infrastructure resources; merging triggers change in Linode*
```
Comment (bibliotechy) [inline infra/ansible-server.md]: We could run something like https://github.com/runatlantis/atlantis that performs the `terrafrom plan` and comments on the PR with the changes that will occur? THoughts?

File: general/ansible-best-practices.md (hunk 1 above)
Comment (cmharlow) [reply general/ansible-best-practices.md]: changed.

File: infra/ansible-server.md (hunk 2 above)
Comment (cmharlow) [reply infra/ansible-server.md]: updated.

File: infra/ansible-server.md (hunk 3 above)
Comment (cmharlow) [reply infra/ansible-server.md]: that's a good idea; adding to issues for filling out around terraform. above i thinking more about how to daisy-chain terraform apply's success to inform ansible's dynamic inventorying & needed reruns. and also trying to figure out a way to separate app deployment from infra deployment.

**Pull request merged**